polaris/viz/planar.py

```diff
@@ -102,37 +128,48 @@ def plot_horiz_field(ds, ds_mesh, field_name, out_file_name=None,  # noqa: C901
         t_index = 0
     if t_index is not None:
         field = field.isel(Time=t_index)
-    if 'nVertLevels' in field.dims and z_index is None:
+    if vert_dim in field.dims and z_index is None:
         z_index = 0
     if z_index is not None:
-        field = field.isel(nVertLevels=z_index)
-
-    if 'nCells' in field.dims:
-        ocean_mask = ds.maxLevelCell - 1 >= 0
-        ocean_patches, ocean_mask = _compute_cell_patches(ds_mesh, ocean_mask)
-    elif 'nEdges' in field.dims:
-        ocean_mask = np.ones_like(field, dtype='bool')
-        ocean_mask = _remove_boundary_edges_from_mask(ds_mesh, ocean_mask)
-        ocean_patches, ocean_mask = _compute_edge_patches(ds_mesh, ocean_mask)
-    ocean_patches.set_array(field[ocean_mask])
+        field = field.isel({vert_dim: z_index})
+
+    if patches is not None:
+        if patch_mask is None:
+            raise ValueError('You must supply both patches and patch_mask '
+                             'from a previous call to plot_horiz_field()')
+    else:
+        if cell_mask is None:
+            cell_mask = np.ones_like(field, type='bool')
```

Comment:
@xylar This needs to be `dtype`. Do you want me to open a separate bugfix PR or do you want to combine this with  https://github.com/E3SM-Project/polaris/pull/144?

Reply:
This should be a separate PR.  I'm not sure how soon #144 will be ready to go in.  Sorry about that.